esp/esdllib/README.md

```diff
@@ -701,9 +787,252 @@ Is not applied to service methods with `@querytype` values:
+
+ESDL Scripts extend standard XPath features to make setting entry point default `@source` and `@target` nodes easier. In the root of any entry point, the `@source` and `@target` XPaths may contain the [Implicit Variables](#implicit-variables) as *Axes*. These special variable references are wrapped in curly braces.
+
+For example, say we have the following simplified request bound for a roxie backend that we're going to manipulate with a `BackendRequest` entry point script:
+
+```xml
+<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
+  <soap:Body>
+    <MyMethodModule.MyMethodQuery>
+      <Context>...</Context>
+      <MyMethodRequest>
+        <Row>
+          <Options></Options>
+        </Row>
+      </MyMethodRequest>
+    </MyMethodModule.MyMethodQuery>
+  </soap:Body>
+</soap:Envelope>
+```
+
+The goal is to have a single script for a service that adds several nodes to the `Options` section for every method in the service. Each method has a different request roxie query name (here `MyMethodModule.MyMethodQuery`) and different method request name (here `MyMethodRequest`). This script will correctly set the default target for all methods:
+
+```xml
+
+<es:BackendRequest name="SetServiceOptions"
+                   target="soap:Body/{$query}/{$request}/Row/Options"
+                   xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/" xmlns:es="urn:hpcc:esdl:script">
+  <es:set-value target="AccessLevel" select="1"/>
+  <es:set-value target="FeatureSet" select="'all'"/>
+</es:BackendRequest>
+```
+
+Resulting in this request sent to the roxie:
+
+```xml
+<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
+  <soap:Body>
+    <MyMethodModule.MyMethodQuery>
+      <Context>...</Context>
+      <MyMethodRequest>
+        <Row>
+          <Options>
+            <AccessLevel>1</AccessLevel>
+            <FeatureSet>1</FeatureSet>
+          </Options>
+        </Row>
+      </MyMethodRequest>
+    </MyMethodModule.MyMethodQuery>
+  </soap:Body>
+</soap:Envelope>
+```
+
+### Namespaces
+
+The scripting engine requires strict adherence to namespaces correctness. A frequent source of scripting errors is omitting namespaces where they're required or using an incorrect namespace or namespace prefix. Namespaces must match between the context XML document and script XPaths referencing the context XML document. It is not required for the prefix in the script's XPath to match the prefix in the document.
+
+In most cases this means that a script needs to define matching namespace URIs. There are two exceptions. The first exception is that the engine will automatically detect and define the standard soap namespace ( `xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/"` ) with the prefix `soap` if you use an XPath that incudes `Body` or `Envelope` nodes. The second exception is that the prefix `n` is defined to refer to the current default namespace, and it can be used in script XPaths in some circumstances.
```

Comment:
I learned something new today. What happens if I explicitly define a namespace prefix `n`? Is `n` reserved or can it be overridden?